lua/harpoon/init.lua

```diff
@@ -51,6 +51,10 @@ function Harpoon:list(name)
     local existing_list = lists[name]
 
     if existing_list then
+        if not self.data.seen[key] then
+            self.data.seen[key] = {}
+        end
+        self.data.seen[key][name] = true
```

Comment:
Since `Data:data()` marks the list as seen, if we find it and early return it won't be marked as seen. Doing so here fixes the issue.